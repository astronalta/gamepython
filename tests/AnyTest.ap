
Test < Object {
}

Constructor < Object {
    
    @init(string String) {
        if string {
            Io::print(string)
        } else {
            Io::print("any\n")
        }
    }
}

Generic[:a] < Object {

    test(obj :a) {
        if obj {
            Io::print("true\n")
        } else {
            Io::print("false\n")
        } 
    }
}

test(string String) {

    if string {
        Io::print(string)
    } else {
        Io::print("any\n")
    }
}

main() Int {
    
    any Any = "hello world\n"
    string String = any

    Io::print(string)

    any = Test()
    string = any

    if string {
        Io::print("fail\n")
    } else {
        Io::print("pass\n")
    }
    

    any2 Any = "hello\n"

    test(any2)
    test(any)

    Constructor(any2)
    Constructor(any)

    g1 = Generic[String]()
    g2 = Generic[Any]()

    g1.test(any2)
    g1.test(any)


    g2.test(any2)
    g2.test(any)

    any = 9
    int Int = any

    Io::print("#{int}\n")

    ret 0
}


