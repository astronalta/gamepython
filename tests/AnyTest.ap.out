hello world
pass
hello
any
hello
any
true
false
true
true
9
test
0
test
9
test
0
any
99
pass
pass
